src/Mvc/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets

```
@@ -0,0 +1,151 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project>
+  <!-- Internal settings. Not intended for customization. -->
+  <PropertyGroup>
+    <GenerateOpenApiReferenceCodeDependsOn>
+      _GetMetadataForOpenApiReferences;
+      _GenerateOpenApiReferenceCode;
+      _CreateCompileItemsForOpenApiReferences
+    </GenerateOpenApiReferenceCodeDependsOn>
+    <GenerateServiceFileReferenceCodeDependsOn>
+      _GenerateErrorsForOldItems;
+      _CreateOpenApiReferenceItemsForOpenApiProjectReferences;
+      GenerateOpenApiReferenceCode
+    </GenerateServiceFileReferenceCodeDependsOn>
+  </PropertyGroup>
+
+  <!-- OpenApiProjectReference support. -->
+
+  <ItemGroup>
+    <!--
+      Do not change %(ReferenceOutputAssembly) if project contains duplicate @(ProjectReference) and
+      @(OpenApiProjectReference) items.
+    -->
+    <ProjectReference Include="@(OpenApiProjectReference)" Exclude="@(ProjectReference)" ReferenceOutputAssembly="false" />
+    <ProjectReference Update="@(OpenApiProjectReference)" OpenApiReference="true" />
+  </ItemGroup>
+
+  <Target Name="_CreateOpenApiReferenceItemsForOpenApiProjectReferences"
+      AfterTargets="ResolveProjectReferences"
+      Condition="'%(ProjectReferenceWithConfiguration.OpenApiReference)' == 'true' AND '$(DesignTimeBuild)' != 'true' AND '$(BuildingProject)' == 'true'"
+      Inputs="%(ProjectReferenceWithConfiguration.Identity)"
+      Outputs="&lt;not-a-file !&gt;">
+    <PropertyGroup>
+      <_Identity>%(ProjectReferenceWithConfiguration.Identity)</_Identity>
+      <_Temporary />
+    </PropertyGroup>
+
+    <MSBuild Target="OpenApiGetDocuments"
+        Project="$(_Identity)"
+        Properties="%(ProjectReferenceWithConfiguration.SetConfiguration); %(ProjectReferenceWithConfiguration.SetPlatform); %(ProjectReferenceWithConfiguration.SetTargetFramework)"
+        RebaseOutputs="true"
+        RemoveProperties="%(ProjectReferenceWithConfiguration.GlobalPropertiesToRemove);TargetFrameworks;RuntimeIdentifier">
+      <Output TaskParameter="TargetOutputs" PropertyName="_Temporary" />
+    </MSBuild>
+
+    <ItemGroup>
+      <OpenApiProjectReference Update="$(_Identity)">
+        <Documents>$(_Temporary)</Documents>
+      </OpenApiProjectReference>
+      <OpenApiReference Include="@(OpenApiProjectReference -> '%(Documents)')" Condition="'%(Identity)' == '$(_Identity)'" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_Identity />
+      <_Temporary />
+    </PropertyGroup>
+  </Target>
+
+  <!-- OpenApiReference support. -->
+
+  <Target Name="_GetMetadataForOpenApiReferences" Condition="'@(OpenApiReference)' != ''">
+    <ItemGroup>
+      <_Temporary Remove="@(_Temporary)" />
+    </ItemGroup>
+
+    <GetFileReferenceMetadata Inputs="@(OpenApiReference)"
+        Extension="$(DefaultLanguageSourceExtension)"
+        Namespace="$(RootNamespace)"
+        OutputDirectory="$(OpenApiDefaultOutputDirectory)">
+      <Output TaskParameter="Outputs" ItemName="_Temporary" />
```

```suggestion
      <Output TaskParameter="Outputs" ItemName="OpenApiReferenceWithMetadata" />
```

A good practice in MSBuild is to use new item groups as you transform the data, instead of updating the items in place. This helps with extensibility because then people extending your targets don't need to worry so much about the order in which targets execute.